### DISCUSSION TITLE
Support for get generic property names with nameof()

### DISCUSSION THREAD

**Discussion by ali50m** [dotnet/csharplang#5164] [Language Ideas] [answered]
Is it able to call nameof method to retrieve the literal name for a generic property?

     ////////////// generic  /////////////////////////////////
     [Fact]
     public void GetPropertyName_FromGeneric()
     {
         // un-compilable
         // Console.WriteLine(nameof(Foo<T>.Value));
         // Console.WriteLine(nameof(Foo<>.Value));
     }
     private class Foo<T>
     {
         public T? Value { get; set; }
     }
     ////////////// no-generic  /////////////////////////////////
     [Fact]
     public void GetPropertyName_FromNonGeneric()
     {
         Console.WriteLine(nameof(Bar.Value));
     }
     private class Bar
     {
         public int Value { get; set; }
     }

**Comment by huoyaoyuan** [comment, marked as answer]
#702
